main.py

```diff
@@ -7,48 +7,38 @@
 
 import dash
 
-from domain.cached_repo import CachedRepo
+from domain.cached_repo_request_driven import CachedRepoRequestDriven
 from infrastructure.config import ConfigYAML
 from infrastructure.data_access.http.synthetics_repo import SyntheticsRepo
 from presentation.main_view import make_page_layout
 from presentation.matrix_view import make_mesh_test_matrix_layout
 
+FORMAT = "%(asctime)-15s  %(message)s"
+logging.basicConfig(level=logging.DEBUG, format=FORMAT)
 logger = logging.getLogger(__name__)
 
 
 class WebApp:
     def __init__(self, email, token: str) -> None:
         try:
+            config = ConfigYAML("config.yaml")
             repo = SyntheticsRepo(email, token)
-            self._config = ConfigYAML("config.yaml")
-            self._cached_repo = CachedRepo(repo, self._config.test_id)
+            self._cached_repo = CachedRepoRequestDriven(
+                repo,
+                config.test_id,
+                config.data_update_period_seconds,
+                config.data_update_lookback_seconds,
+            )
+            self._config = config
         except Exception as err:
-            logger.exception(f"WebApp initialization failure: {err}")
+            logger.exception("WebApp initialization failure")
```

Comment:
Won't this change filter the root cause of the error?

Reply:
Logger.exception attaches current exception to the log